tools/ci/check_c_api_usage.py

```diff
@@ -0,0 +1,211 @@
+            continue
+
+        # Not in block: look for next // or /* from current i
+        sl = line.find("//", i)
+        bl = line.find("/*", i)
+
+        if sl != -1 and (bl == -1 or sl < bl):
+            # Line comment starts first: take code up to '//' and stop
+            out_parts.append(line[i:sl])
+            return ("".join(out_parts), in_block)
+
+        if bl != -1:
+            # Block comment starts: take code up to '/*', then enter block
+            out_parts.append(line[i:bl])
+            i = bl + 2
+            in_block = True
+            continue
+
+        # No more comments
+        out_parts.append(line[i:])
+        break
+
+    return ("".join(out_parts), in_block)
+
+def iter_source_files(root: Path, exts: set[str], excludes: set[str]) -> list[Path]:
+    """
+    Return a list of source files under 'root', where filenames end with any of the
+    extensions in 'exts' (e.g., '.c.src', '.c', '.h').
+    Excludes directories whose names are in 'excludes'.
+    """
+    results: list[Path] = []
+
+    for dirpath, dirnames, filenames in os.walk(root):
+        # Prune excluded directories
+        dirnames[:] = [d for d in dirnames if d not in excludes]
+        for fn in filenames:
+            # endswith handles mult-suffice patterns, e.g., .c.src
+            if any(fn.endswith(ext) for ext in exts):
+                results.append(Path(dirpath) / fn)
+    return results
+
+
+def scan_file(path: Path) -> list[tuple[str, int, str, str]]:
+    """
+    Scan a single file.
+    Returns list of (func_name, line_number, path_str, raw_line_str).
+    """
+    hits: list[tuple[str, int, str, str]] = []
+    in_block = False
+
+    try:
+        with path.open("r", encoding="utf-8", errors="ignore") as f:
+            for lineno, raw in enumerate(f, 1):
+                # Skip if approved by noqa markers
+                if NOQA_OK in raw or NOQA_MANUAL in raw:
+                    continue
+
+                # Remove comments; if nothing remains, skip
+                code, in_block = strip_comments(raw.rstrip("\n"), in_block)
+                if not code.strip():
+                    continue
+
+                # Find all suspicious calls in non-comment code
+                for m in FUNC_RX.finditer(code):
+                    func = m.group(0)
+                    hits.append((func, lineno, str(path), raw.rstrip("\n")))
+    except FileNotFoundError:
+        # File may have disappeared; ignore gracefully
+        pass
+    return hits
+
+
+def main(argv: list[str] | None = None) -> int:
+    ap = argparse.ArgumentParser(description="Borrow-ref C API linter (Python).")
+    ap.add_argument(
+        "--root",
+        default="numpy",
+        type=str,
+        help="Root directory to scan (default: numpy)"
+        )
+    ap.add_argument(
+        "--ext",
+        action="append",
+        default=None,
+        help="File extension(s) to include (repeatable). Defaults to .c,.h,.c.src,.cpp",
```

Comment:
I wonder if it would make sense to just substitute the variable in directly here to avoid hardcoding the defaults in two separate locations, i.e.,

```diff
--- a/tools/ci/check_c_api_usage.py
+++ b/tools/ci/check_c_api_usage.py
@@ -148,7 +148,7 @@ def main(argv: list[str] | None = None) -> int:
         "--ext",
         action="append",
         default=None,
-        help="File extension(s) to include (repeatable). Defaults to .c,.h,.c.src,.cpp",
+        help=f"File extension(s) to include (repeatable). Defaults to {DEFAULT_EXTS}",
```